### ISSUE TITLE
Rule Change: no-implicit-globals lexicalBindings without functions

### ISSUE THREAD

**Issue by kkmuffme** [eslint/eslint#16328]
### What rule do you want to change?

no-implicit-globals

### What change to do you want to make?

Implement suggestions

### How do you think the change should be implemented?

A new option

### Example code

```js
/**
 * @returns {int}
 */
function testFunc() {
    "use strict";

    return 1 + 2;
}
```

### What does the rule currently do for this code?

When using with lexicalBindings `/*eslint no-implicit-globals: ["error", {"lexicalBindings": true}]*/` it gives an error

### What will the rule do after it's changed?

Not report an error for functions in global context.
e.g. if there was an option for that like:
`/*eslint no-implicit-globals: ["error", {"lexicalBindingsVars": true}]*/`

to only apply it to const/let, or alternatively a different value ('only-variables' or 'no-functions'):
`/*eslint no-implicit-globals: ["error", {"lexicalBindings": "no-functions"}]*/`
or
`/*eslint no-implicit-globals: ["error", {"lexicalBindings": "only-variables"}]*/`

**Comment by mdjermanovic** [comment]
The `lexicalBindings` option doesn't relate to functions. This rule checks and reports functions by default, regardless of this option, here's an example in [eslint playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWltcGxpY2l0LWdsb2JhbHM6IFwiZXJyb3JcIiAqL1xuXG4vKipcbiAqIEByZXR1cm5zIHtpbnR9XG4gKi9cbmZ1bmN0aW9uIHRlc3RGdW5jKCkge1xuICAgIFwidXNlIHN0cmljdFwiO1xuXG4gICAgcmV0dXJuIDEgKyAyO1xufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnsiZXM2Ijp0cnVlfX19). The `lexicalBindings` option is intended to include lexical bindings (not to exclude others), and was added for backward compatibility reasons only. This rule should actually always report all declarations that create global variables, and I don't think there should be an option to exclude _all_ function declarations. I'm more in favor of marking individual declarations as intentionally global, as proposed in https://github.com/eslint/eslint/issues/16327.
